oracle/pkg/updater/updater.go

```diff
@@ -460,15 +468,43 @@ func (u *Updater) getL1Txns(ctx context.Context, blockNum uint64) (map[string]in
 
 	u.metrics.BlockTxnCacheMisses.Inc()
 
-	blk, err := u.l1Client.BlockByNumber(ctx, big.NewInt(0).SetUint64(blockNum))
+	block, err := u.l1Client.BlockByNumber(ctx, big.NewInt(0).SetUint64(blockNum))
 	if err != nil {
 		return nil, fmt.Errorf("failed to get block by number: %w", err)
 	}
 
-	txnsInBlock := make(map[string]int)
-	for posInBlock, tx := range blk.Transactions() {
-		txnsInBlock[strings.TrimPrefix(tx.Hash().Hex(), "0x")] = posInBlock
+	txnsInBlock := make(map[string]TxMetadata)
+	var wg sync.WaitGroup
+	var mu sync.Mutex
+	var receiptErr error
+
+	processTransactionMetadata := func(posInBlock int, tx *types.Transaction) {
+		defer wg.Done()
+		receipt, err := u.l1Client.TransactionReceipt(ctx, tx.Hash())
```

Comment:
Yes, I didn't realize we had this.

Reply:
Changing this will require significantly more time, as we need to implement the TxMonitor interface across the EVMClient. Seems like a non-trivial amount of refactoring for something that won't necessiarly be a large overhead? Also are we sure this is the critical path? It's simply getting the txn receipts, which we do quite rarely.